optuna/storages/journal/_storage.py

```diff
@@ -266,13 +266,13 @@ def create_new_trial(self, study_id: int, template_trial: FrozenTrial | None = N
             self._sync_with_backend()
             trial_id = self._replay_result._last_created_trial_id_by_this_process
 
-        # Dump snapshot here.
-        if (
-            isinstance(self._backend, BaseJournalSnapshot)
-            and trial_id != 0
-            and trial_id % SNAPSHOT_INTERVAL == 0
-        ):
-            self._backend.save_snapshot(pickle.dumps(self._replay_result))
+            # Dump snapshot here.
+            if (
+                isinstance(self._backend, BaseJournalSnapshot)
+                and trial_id != 0
+                and trial_id % SNAPSHOT_INTERVAL == 0
+            ):
+                self._backend.save_snapshot(pickle.dumps(self._replay_result))
```

Comment:
`backend.save_snapshot` takes a relatively long time. How about acquiring the lock object while only calling `pickle.dumps`?

```suggestion
        # Dump snapshot here.
        if (
            isinstance(self._backend, BaseJournalSnapshot)
            and trial_id != 0
            and trial_id % SNAPSHOT_INTERVAL == 0
        ):
            with self._thread_lock:
                replay_result_pkl = pickle.dumps(self._replay_result)
            self._backend.save_snapshot(replay_result_pkl)
```

Reply:
I think it's reasonable, but other parts of the code are pessimistic lock.

So I think it's better to follow the other code in this PR.